Assets/_BForBoss/_Core/Scripts/EnergySystem/Data/EnergySO.cs

```diff
@@ -0,0 +1,25 @@
+using UnityEngine;
+
+namespace BForBoss
+{
+    [CreateAssetMenu(fileName = "Energy Data", menuName = "PerigonGames/EnergySystem/EnergyData", order = 1)]
+    public class EnergySO : ScriptableObject
+    {
+        [SerializeField, Tooltip("Player's current value of Energy"), Min(0f)] 
+        private float _value = 0f;
+
+        [SerializeField, Tooltip("Max Energy Attainable")]
+        private float _maxEnergyValue = 100f;
+
+        [SerializeField, Tooltip("How often does the player gain energy per second when in the middle of accruement")]
+        private float _rateOfAccruement = 1f;
```

Comment:
Should we also have a `_rateOfExpense` field since things like SlowMo is an expense that also goes at a rate

Reply:
Hmmm I think this rate should be for both Accurument and Expense. 
To keep it consistent between both, and not expose too many dynamic variable (keep it simple). 

So this should be called `_rateOfChange` or if you have a better name that encomposses either accrument or expense

Reply:
Will call it `_rateOfTransaction` to keep in line with our documentation I guess!